packages/opentelemetry-core/src/utils/environment.ts

```diff
@@ -64,6 +64,28 @@ function parseNumber(
   }
 }
 
+/**
+ * Parses a LogLevelString as a LogLevel
+ * @param key
+ * @param environment
+ * @param values
+ */
+function parseLogLevel(
```

Comment:
This function seems quite convoluted. Why not just have a switch/case?

Reply:
Is this function modifying the environment? Why not a function that just returns a log level and use that?

Reply:
switch/case sounds good, I was trying to be clever and get the values of logLevelStrings without repeating them (`DEBUG`, `INFO`, `WARN` and `ERROR`) but it's too much convolution as mentioned.

Reply:
`['DEBUG', 'INFO', ....].indexOf(checkMe)>=0` isn't enough ?

Reply:
I went with the switch case for now, 

@obecny I do not fully understand this suggestion could you please explain?

Reply:
was just referring to validation with regexp etc., loglevel can have 4 values only